### DISCUSSION TITLE
Revert secrets sanitize exception

### DISCUSSION THREAD

**Discussion by xoxys** [woodpecker-ci/woodpecker#3197] [General]
@woodpecker-ci/maintainers I'm a bit confused about this decision made in https://github.com/woodpecker-ci/woodpecker/pull/2680, and I would like to discuss it:

- Why `<=3`? Is there any resilient basis that secrets with `<=3` chars are weak in terms of entropy? Secrets with `<=4` chars are not weak anymore? This sounds a bit random to me.
- Even if such secrets are weak, IMO we make a personal assumption here. There might still be valid reasons for users to "hide" a 3 char value in a secret, even if it is weak and we should not print it.

I would vote for reverting this PR and sanitize all secrets regardless of their length. I came across this while working on https://github.com/woodpecker-ci/woodpecker/pull/3171#issuecomment-1887663999 where I tried so understand some "magic numbers".

**Comment by anbraten** [comment]
The length comes from the fact that 3 or more chars words are pretty unique in texts already. Using a 1 char secret is fine for WP, but hiding it in logs makes no sense as you would already now which char it is from reading a log like this and IMO it just looks pretty weird as well: `the r*d fox jumps ov*r th* lazy brown dog`. Having false positive matches with 3 chars or more would be just less likely.

**Reply by xoxys** [reply]
Thanks for taking the time to explain it. That makes a lot more sense now.

**Comment by qwerty287** [comment]
@xoxys Do you still think it should be reverted or can we close this?
